api/v1alpha3/azuremachine_conversion.go

```diff
@@ -45,6 +45,9 @@ func (src *AzureMachine) ConvertTo(dstRaw conversion.Hub) error { // nolint
 		}
 	}
 
+	if restored.Spec.DataDisks != nil {
+		dst.Spec.DataDisks = restored.Spec.DataDisks
```

Comment:
sorry I just saw this, I think this would override any changes that were made to the Disk between the last conversion and now, we only want to restore UltraSSD disks . For example, let's say you create a data disk with another storage type, convert the Machine to v1alpha3, then delete the disk, then convert it back to v1alpha4. This line would re-add the deleted disk, which is incorrect. We only want to restore fields that were not supported in v1alpha3 when converting to v1alpha4.

Reply:
here's an example you can use: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/api/v1alpha3/azurecluster_conversion.go#L66

Reply:
this is the commit for reference https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/715d5fd7363c92713086a0304152a3c3612315d8#diff-513d36576b5dbcc8430d6b8a26b0f7cccda060c2d932cc1a0d38f7dc16a9e03e